tensorflow/core/kernels/mkl_matmul_op_fused.cc

```diff
@@ -0,0 +1,192 @@
+
+#include "tensorflow/core/framework/register_types.h"
+#include "tensorflow/core/kernels/fill_functor.h"
+#include "tensorflow/core/kernels/mkl_matmul_ops_common.h"
+#include "tensorflow/core/lib/core/errors.h"
+
+namespace tensorflow {
+
+// Fuse Operation
+template <typename Device, typename T>
+class MklFusedMatMulOp : public MklDnnMatMulOpBase<T> {
+ public:
+  explicit MklFusedMatMulOp(OpKernelConstruction* ctx)
+      : MklDnnMatMulOpBase<T>(ctx) {
+    std::vector<string> fused_ops;
+
+    OP_REQUIRES_OK(ctx, ctx->GetAttr("fused_ops", &fused_ops));
+    OP_REQUIRES_OK(ctx, ctx->GetAttr("transpose_a", &transpose_a_));
+    OP_REQUIRES_OK(ctx, ctx->GetAttr("transpose_b", &transpose_b_));
+
+    OP_REQUIRES(ctx, fused_ops.size() == 1,
+                errors::InvalidArgument(
+                    "MklFusedMatMul must have only one argument: bias."));
+    OP_REQUIRES(
+        ctx, transpose_a_ == false,
+        errors::InvalidArgument("In[0] of MklMatMul can't be transposed."));
+  }
+
+  void Compute(OpKernelContext* ctx) override {
+    // FusedMatMul has 3 inputs: src, weights, bias
+    const Tensor& src_tensor = ctx->input(this->kInputIndexSrc);
+    const Tensor& weight_tensor = ctx->input(this->kInputIndexWeight);
+    const Tensor& bias_tensor = MklGetInput(ctx, this->kInputIndexBias);
+
+    MklDnnShape src_mkl_shape;
+    MklDnnShape weight_mkl_shape;
+    GetMklShape(ctx, this->kInputIndexSrc, &src_mkl_shape);
+    GetMklShape(ctx, this->kInputIndexWeight, &weight_mkl_shape);
+
+    // Get shapes of input tensors
+    auto src_tf_shape = src_mkl_shape.IsMklTensor() ? src_mkl_shape.GetTfShape()
+                                                    : src_tensor.shape();
+    auto weight_tf_shape = weight_mkl_shape.IsMklTensor()
+                               ? weight_mkl_shape.GetTfShape()
+                               : weight_tensor.shape();
+
+    // Check the constraint of input matrix and bias
+    OP_REQUIRES(ctx, TensorShapeUtils::IsMatrix(src_tf_shape),
+                errors::InvalidArgument("In[0] is not a matrix"));
+    OP_REQUIRES(ctx, TensorShapeUtils::IsMatrix(weight_tf_shape),
+                errors::InvalidArgument("In[1] is not a matrix"));
+    OP_REQUIRES(ctx, TensorShapeUtils::IsVector(bias_tensor.shape()),
+                errors::InvalidArgument("Biases must be 1D"));
+
+    // Expression: [batch, k] * [k, channel] + [channel] = [batch, channel]
+    //
+    // Get dimension size of each matrix, dim_pair[] is the location of k
+    // in the inputs, we have constraint that k of the two inputs are
+    // the same
+    const int dim_pair[] = {1, transpose_b_ ? 1 : 0};
+    const int batch = src_tf_shape.dim_size(1 - dim_pair[0]);
+    const int k = src_tf_shape.dim_size(dim_pair[0]);
+    const int channel = weight_tf_shape.dim_size(1 - dim_pair[1]);
+
+    OP_REQUIRES(ctx, k == weight_tf_shape.dim_size(dim_pair[1]),
+                errors::InvalidArgument("Matrix size are incompatible: In[0]: ",
```

Comment:
```suggestion
                errors::InvalidArgument("Matrix sizes are incompatible: In[0]: ",
```

Reply:
I've reverted the change here because there's a case checking the error verbose like this: https://github.com/tensorflow/tensorflow/blob/master/tensorflow/python/eager/function_test.py#L270